cnxpublishing/views/status.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+# ###
+# Copyright (c) 2013-2016, Rice University
+# This software is subject to the provisions of the GNU Affero General
+# Public License version 3 (AGPLv3).
+# See LICENCE.txt for details.
+# ###
+from __future__ import absolute_import
+
+import psycopg2
+from celery.result import AsyncResult
+from pyramid import httpexceptions
+from pyramid.view import view_config
+
+from .. import config
+
+
+@view_config(route_name='print-style-history', request_method='GET',
+             renderer='json')
+def print_style_history(request):
+    settings = request.registry.settings
+    db_conn_str = settings[config.CONNECTION_STRING]
+
+    styles = []
+    with psycopg2.connect(db_conn_str) as db_conn:
+        with db_conn.cursor() as cursor:
+            cursor.execute("""
+            SELECT print_style, revised, fileid, tag
+            FROM print_style_recipes
+            ORDER BY revised DESC;
+            """)
+            response = cursor.fetchall()
+            for row in response:
+                styles.append({
+                    'print_style': row[0],
+                    'revised': str(row[1]),
+                    'recipe': row[2],
+                    'version': row[3],
+                    'print_style_url': request.route_url(
+                        'print-style-history-name',
+                        name=row[0]),
+                    'recipe_url': request.route_url(
+                        'print-style-history-version',
+                        name=row[0], version=row[3])
+                })
+    return styles
+
+
+@view_config(route_name='print-style-history_POST', request_method='POST',
+             renderer='json')
+def print_style_history_POST(request):
+    settings = request.registry.settings
+    db_conn_str = settings[config.CONNECTION_STRING]
+
+    files = request.POST.get('files')
+    f = files['recipe']
+    data = request.POST.get('data')
+    name = data['name']
+    version = data['version']
```

Comment:
Okay, So just have a POST that gets those three fields. I think I got confused because this line makes a POST request and sends these variables. I think I get it now though. :) https://github.com/Connexions/cnx-deploy/blob/print-styles/roles/_content_helpers/library/upload_recipes.py#L29